aiohttp/web.py

```diff
@@ -491,6 +490,7 @@ def run_app(
             access_log.addHandler(logging.StreamHandler())
 
     try:
+        asyncio.set_event_loop(loop)
```

Comment:
I don't think it's a good idea to have it here. At least, put it right before `try:`. The reason is that it's an antipattern to have more than one instruction in a try block (it tends to shadow exceptions sometimes and causes hard-to-debug situations).

Reply:
Yeah, I just noticed that it was inside the try block in `asyncio.run()`, figured there was probably a good reason for it.

Reply:
Well, you could add a separate try/except if you want to call `loop.close()` and re-raise. There's no need to `asyncio.set_event_loop(None)` because setting it didn't happen anyway.

Reply:
I've kept these 2 lines together in the try to match https://github.com/python/cpython/blob/main/Lib/asyncio/runners.py#L40-L44

But, I've moved the create_task() to before the try statement, is it doesn't seem to have any business being there.